packages/flutter/lib/src/gestures/resampler.dart

```diff
@@ -0,0 +1,422 @@
+        buttons: event.buttons,
+        obscured: event.obscured,
+        pressure: event.pressure,
+        pressureMin: event.pressureMin,
+        pressureMax: event.pressureMax,
+        distanceMax: event.distanceMax,
+        size: event.size,
+        radiusMajor: event.radiusMajor,
+        radiusMinor: event.radiusMinor,
+        radiusMin: event.radiusMin,
+        radiusMax: event.radiusMax,
+        orientation: event.orientation,
+        tilt: event.tilt,
+        embedderId: event.embedderId,
+      );
+    }
+    if (event is PointerMoveEvent) {
+      return PointerMoveEvent(
+        timeStamp: timeStamp,
+        pointer: pointerIdentifier,
+        kind: event.kind,
+        device: event.device,
+        position: position,
+        delta: delta,
+        buttons: event.buttons,
+        obscured: event.obscured,
+        pressure: event.pressure,
+        pressureMin: event.pressureMin,
+        pressureMax: event.pressureMax,
+        distanceMax: event.distanceMax,
+        size: event.size,
+        radiusMajor: event.radiusMajor,
+        radiusMinor: event.radiusMinor,
+        radiusMin: event.radiusMin,
+        radiusMax: event.radiusMax,
+        orientation: event.orientation,
+        tilt: event.tilt,
+        platformData: event.platformData,
+        synthesized: event.synthesized,
+        embedderId: event.embedderId,
+      );
+    }
+    if (event is PointerUpEvent) {
+      return PointerUpEvent(
+        timeStamp: timeStamp,
+        pointer: pointerIdentifier,
+        kind: event.kind,
+        device: event.device,
+        position: position,
+        buttons: event.buttons,
+        obscured: event.obscured,
+        pressure: event.pressure,
+        pressureMin: event.pressureMin,
+        pressureMax: event.pressureMax,
+        distance: event.distance,
+        distanceMax: event.distanceMax,
+        size: event.size,
+        radiusMajor: event.radiusMajor,
+        radiusMinor: event.radiusMinor,
+        radiusMin: event.radiusMin,
+        radiusMax: event.radiusMax,
+        orientation: event.orientation,
+        tilt: event.tilt,
+        embedderId: event.embedderId,
+      );
+    }
+    if (event is PointerCancelEvent) {
+      return PointerCancelEvent(
+        timeStamp: timeStamp,
+        pointer: pointerIdentifier,
+        kind: event.kind,
+        device: event.device,
+        position: position,
+        buttons: event.buttons,
+        obscured: event.obscured,
+        pressureMin: event.pressureMin,
+        pressureMax: event.pressureMax,
+        distance: event.distance,
+        distanceMax: event.distanceMax,
+        size: event.size,
+        radiusMajor: event.radiusMajor,
+        radiusMinor: event.radiusMinor,
+        radiusMin: event.radiusMin,
+        radiusMax: event.radiusMax,
+        orientation: event.orientation,
+        tilt: event.tilt,
+        embedderId: event.embedderId,
+      );
+    }
+    assert(event is PointerRemovedEvent);
```

Comment:
Could a `PointerScrollEvent` reach here and trigger the assert?

Reply:
Good point. I've updated the documentation to make it clear what event types are supported and gesture binding is not limited to touch events.